stdlib/queue.gr

```diff
@@ -295,4 +327,43 @@ provide module Immutable {
       { forwards, backwards } => List.length(forwards) + List.length(backwards),
     }
   }
+
+  /**
+   * Converts a queue into a list of its elements.
+   *
+   * @param queue: The queue to convert
+   * @returns A list containing all queue values
+   *
+   * @since v0.6.0
+   */
+
+  provide let toList = queue => {
+    let mut queue = queue
+    let mut lst = []
+    while (!isEmpty(queue)) {
+      match (peek(queue)) {
+        Some(value) => {
+          queue = pop(queue)
+          lst = [value, ...lst]
+        },
+        None => break,
+      }
+    }
+    List.reverse(lst)
+  }
+
+  /**
+   * Creates a queue from a list.
+   *
+   * @param list: The list to convert
+   * @returns A queue containing all list values
+   *
+   * @since v0.6.0
+   */
+
+  provide let fromList = list => {
+    let mut queue = empty
+    List.forEach(e => queue = push(e, queue), list)
+    queue
```

Comment:
Given the above, can't this also be implemented by just constructing the record with the list as `forwards` ?